File: create-favicons.py
#!/usr/bin/env python3
"""
Create ILS-branded favicon PNG files
Simple script using PIL/Pillow
"""

try:
    from PIL import Image, ImageDraw, ImageFont
    import os

    def create_ils_favicon(size, output_path):
        """Create an ILS-branded favicon"""
        # Create image with dark blue background
        img = Image.new('RGBA', (size, size), (15, 23, 42, 255))  # #0f172a
        draw = ImageDraw.Draw(img)

        # Draw lens circles at top
        center_x = size // 2
        center_y = int(size * 0.3)
        outer_radius = int(size * 0.12)
        inner_radius = int(size * 0.08)

        # Outer circle
        draw.ellipse(
            [(center_x - outer_radius, center_y - outer_radius),
             (center_x + outer_radius, center_y + outer_radius)],
            outline=(96, 165, 250, 255),  # #60a5fa
            width=int(size * 0.03)
        )

        # Inner circle
        draw.ellipse(
            [(center_x - inner_radius, center_y - inner_radius),
             (center_x + inner_radius, center_y + inner_radius)],
            outline=(96, 165, 250, 255),  # #60a5fa
            width=int(size * 0.02)
        )

        # Draw ILS text
        try:
            # Try to use a system font
            font_size = int(size * 0.4)
            font = ImageFont.truetype("/System/Library/Fonts/Supplemental/Arial Bold.ttf", font_size)
        except:
            # Fallback to default font
            font = ImageFont.load_default()

        text = "ILS"
        # Get text bounding box
        bbox = draw.textbbox((0, 0), text, font=font)
        text_width = bbox[2] - bbox[0]
        text_height = bbox[3] - bbox[1]

        # Center the text
        text_x = (size - text_width) // 2
        text_y = int(size * 0.65) - text_height // 2

        draw.text(
            (text_x, text_y),
            text,
            fill=(59, 130, 246, 255),  # #3b82f6
            font=font
        )

        # Save
        img.save(output_path, 'PNG')
        print(f"✅ Created {output_path} ({size}x{size})")

    # Create all sizes
    print("Creating ILS-branded favicons...\n")
    create_ils_favicon(128, 'client/public/favicon.png')
    create_ils_favicon(192, 'client/public/icon-192.png')
    create_ils_favicon(512, 'client/public/icon-512.png')
    print("\n✅ All favicons created successfully!")

except ImportError:
    print("❌ Pillow (PIL) not installed")
    print("\nInstall with: pip install Pillow")
    print("Or: pip3 install Pillow")
    print("\nAlternatively, use an online favicon generator:")
    print("1. Go to https://favicon.io/favicon-generator/")
    print("2. Use these settings:")
    print("   - Text: ILS")
    print("   - Background: #0f172a (dark blue)")
    print("   - Font Color: #3b82f6 (blue)")
    print("   - Font: Inter or Arial Bold")
    print("   - Size: Round")
    print("3. Download and extract to client/public/")

File: create-favicons.mjs
#!/usr/bin/env node
/**
 * Create ILS-branded favicon PNG files
 * Simple colored squares with ILS branding
 */

import fs from 'fs';
import { createCanvas } from 'canvas';

function createFavicon(size, outputPath) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  // Background - dark blue
  ctx.fillStyle = '#0f172a';
  ctx.fillRect(0, 0, size, size);

  // Add border radius effect with clipping
  ctx.save();
  const radius = size * 0.15;
  ctx.beginPath();
  ctx.moveTo(radius, 0);
  ctx.lineTo(size - radius, 0);
  ctx.quadraticCurveTo(size, 0, size, radius);
  ctx.lineTo(size, size - radius);
  ctx.quadraticCurveTo(size, size, size - radius, size);
  ctx.lineTo(radius, size);
  ctx.quadraticCurveTo(0, size, 0, size - radius);
  ctx.lineTo(0, radius);
  ctx.quadraticCurveTo(0, 0, radius, 0);
  ctx.closePath();
  ctx.clip();
  ctx.fillStyle = '#0f172a';
  ctx.fillRect(0, 0, size, size);

  // Draw lens circles (top)
  const centerX = size / 2;
  const centerY = size * 0.3;
  const outerRadius = size * 0.12;
  const innerRadius = size * 0.08;

  ctx.strokeStyle = '#60a5fa';
  ctx.lineWidth = size * 0.03;
  ctx.beginPath();
  ctx.arc(centerX, centerY, outerRadius, 0, Math.PI * 2);
  ctx.stroke();

  ctx.lineWidth = size * 0.02;
  ctx.beginPath();
  ctx.arc(centerX, centerY, innerRadius, 0, Math.PI * 2);
  ctx.stroke();

  // Draw ILS text
  const fontSize = size * 0.4;
  ctx.fillStyle = '#3b82f6';
  ctx.font = `bold ${fontSize}px Inter, system-ui, sans-serif`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText('ILS', centerX, size * 0.65);

  ctx.restore();

  // Save to file
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(outputPath, buffer);
  console.log(`✅ Created ${outputPath} (${size}x${size})`);
}

// Check if canvas module is available
try {
  // Create favicons
  createFavicon(128, 'client/public/favicon.png');
  createFavicon(192, 'client/public/icon-192.png');
  createFavicon(512, 'client/public/icon-512.png');
  console.log('\n✅ All favicons created successfully!');
} catch (error) {
  if (error.code === 'MODULE_NOT_FOUND') {
    console.error('❌ canvas module not found');
    console.log('Creating simple fallback icons...\n');

    // Fallback: Create simple solid color PNGs using minimal PNG encoding
    // This is a simplified approach - just update the existing files manually
    console.log('Please install canvas module: npm install canvas');
    console.log('Or use an online tool to create PNG favicons from the SVG');
  } else {
    console.error('Error creating favicons:', error);
  }
}
